experimental/packages/opentelemetry-sdk-metrics-base/src/aggregator/LastValue.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Sum, AggregatorKind, Aggregator, Accumulation, AccumulationRecord } from './types';
+import { HrTime } from '@opentelemetry/api';
+import { InstrumentationLibrary } from '@opentelemetry/core';
+import { Resource } from '@opentelemetry/resources';
+import { PointDataType, SingularMetricData } from '../export/MetricData';
+import { AggregationTemporality } from '@opentelemetry/api-metrics-wip';
+import { InstrumentDescriptor } from '../InstrumentDescriptor';
+import { Maybe } from '../utils';
+
+export class LastValueAccumulation implements Accumulation {
```

Comment:
Can you explain why there is a separate accumulation and aggregation? Does this come from the spec, from emulating another SIG, or from your own idea?

Reply:
According to the spec the last value aggregator should also keep a timestamp of the last value https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#last-value-aggregation

Reply:
> Can you explain why there is a separate accumulation and aggregation

The main issue is that the owner of `Aggregator` (here it is `DeltaMetricStorage`) have to create new state of aggregation (or say accumulation, they know how to aggregate metric values like Sum, LastValue and Histogram) and reset them when the data is being reported. When creating new state of the aggregation, the owner of `Aggregator` must not call `Aggregation.createAggregator(instrumentDescriptor)` again since it should be one single shot on the creation of the metric instrument.

So the role of those three components are:
- `Aggregation` is responsible for selecting concrete `Aggregator` type for an `InstrumentDescriptor`.
- `Aggregator` is stateless, and is responsible for merging, diffing a concrete type (like Sum and Histogram) of stateful accumulations.
- `Accumulation` is responsible for aggregating values (with concrete type like Sum and Histogram) and keep an aggregated state for a metric stream in a period of time.

The spec didn't explain in detail on how `Aggregation` works and it use the term `Aggregation` and `Aggregator` in an interchangeable way confusingly. I followed implementation pattern of java's implementation.

Reply:
We talked about that at the spec call. The java implementation apparently has confusing names because they were brought over from a previous implementation. Because these types are all internal, we can name them what makes sense to us. I believe they want the term `Aggregator` to be reserved from public export so that it can be used in the future by users to make custom aggregators.

Reply:
It is still unclear about the difference between aggregation selector and aggregation in the spec. I posted a comment about it: https://github.com/open-telemetry/opentelemetry-specification/issues/2149#issuecomment-977459635